# Patch needed for ckanext-security

# The server-side session storage requires the session middleware in CKAN
# core to be moved near the end of the middleware stack
# see: https://github.com/data-govt-nz/ckanext-security#requirements
--- a/ckan/config/middleware/flask_app.py
+++ b/ckan/config/middleware/flask_app.py
@@ -205,7 +205,6 @@
             data_dir=cache_dir)

     app.wsgi_app = RootPathMiddleware(app.wsgi_app)
-    app.wsgi_app = SessionMiddleware(app.wsgi_app, session_opts)
     app.session_interface = BeakerSessionInterface()

     # Add Jinja2 extensions and filters
@@ -344,6 +343,9 @@
         who_parser.remote_user_key
     )

+    # Moved as part of ckanext-security patch
+    app = SessionMiddleware(app, session_opts)
+
     # Update the main CKAN config object with the Flask specific keys
     # that were set here or autogenerated
     flask_config_keys = set(flask_app.config.keys()) - set(config.keys())
